core/src/sumeragi/main_loop.rs

```diff
@@ -911,7 +921,21 @@ pub(crate) fn run<F: FaultInjection>(
             },
             Role::ObservingPeer => match message {
                 Some(Message::BlockCreated(BlockCreated { block })) => {
-                    if let Some(block) = vote_for_block(sumeragi, &state, block) {
+                    if current_view_change_index >= 1 {
+                        if let Some(block) = vote_for_block(sumeragi, &state, block) {
+                            let block_hash = block.block.hash();
+
+                            let msg = MessagePacket::new(
+                                view_change_proof_chain.clone(),
+                                BlockSigned::from(block.block.clone()),
+                            );
+
+                            sumeragi.broadcast_packet_to(msg, [current_topology.proxy_tail()]);
+                            info!(%addr, %block_hash, "Block validated, signed and forwarded");
+
+                            voting_block = Some(block);
+                        }
+                    } else if let Some(block) = vote_for_block(sumeragi, &state, block) {
                         voting_block = Some(block);
                     }
```

Comment:
Consider small refactoring: 

```suggestion
                    if let Some(block) = vote_for_block(sumeragi, &state, block) {
                        if current_view_change_index >= 1 {
                            let block_hash = block.block.hash();

                            let msg = MessagePacket::new(
                                view_change_proof_chain.clone(),
                                BlockSigned::from(block.block.clone()),
                            );

                            sumeragi.broadcast_packet_to(msg, [current_topology.proxy_tail()]);
                            info!(%addr, %block_hash, "Block validated, signed and forwarded");
                        }
                        
                        voting_block = Some(block);
                    }
```

Reply:
good suggestion